middleware/recovery.go

```diff
@@ -0,0 +1,129 @@
+package middleware
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"runtime/debug"
+	"sync"
+
+	"github.com/danielgtaylor/huma"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+)
+
+// MaxLogBodyBytes logs at most this many bytes of any request body during a
+// panic when using the recovery middleware. Defaults to 10KiB.
+var MaxLogBodyBytes int64 = 10 * 1024
+
```

Comment:
This feels like the kind of value we may want to make configurable per request, but right now it's a one-time set-and-forget.

Reply:
You can actually set this because it's a public var. You can do `middleware.MaxLogBodyBytes = 10 * 1024 * 1024` to log 10MB before your service starts.

Reply:
Ah, gotcha. I'm still not used to Go's scoping algorithm. "Capitalize to make public" really messes with me. xvx

Reply:
Yeah it's definitely something to get used to :smile: